src/amms/uniswap_v2/mod.rs

```diff
@@ -173,6 +174,39 @@ impl AutomatedMarketMaker for UniswapV2Pool {
         let price = self.calculate_price_64_x_64(base_token)?;
         q64_to_float(price)
     }
+
+    async fn init<T, N, P>(mut self, block_number: u64, provider: Arc<P>) -> Result<Self, AMMError>
```

Comment:
Same here, would be nice to be able to pass `BlockNumberOrTag::Latest` as that will probably be the default functionality